cypress-io/cypress
# feat: debug page tutorial slideshow

- Closes #25768

### Additional details
Adds a slideshow tutorial to the empty states within the Debug page. This will show for three state:
- Not logged in
- Logged in without a connected project
- Logged in with a connected project that has no runs

Upon completing the slideshow (hitting "done"), the default empty state will become the default state again for that project as we save that the user has seen the slideshow in local project settings.

### Steps to test
The `DebugContainer.cy.tsx` spec runs through all states where the slideshow will appear:
https://github.com/cypress-io/cypress/blob/5240290363ddd41f46c81aa34cacd23e8407f49a/packages/app/src/debug/DebugContainer.cy.tsx#L14

To test in a real app:
1.  Check out this branch and run `yarn`
2. Setup a new project (or remove the `projectId` from an existing project) and run `yarn cypress:open --project <path-to-project>`
3. Navigate to the app and click on the `Debug` Tab
4. The slideshow will show up. Refresh the page and the slideshow will still appear.
5. Click through the slideshow until you move past the `Done` tab. The default empty state will now show. Refresh the page and this will now be the default screen.

When you view the slideshow, a cohort will be set (either "A" or "B"). This determines what images to display (skeleton or text). To reset this or change the cohort, open Launchpad and click `Developer Tools -> View App Data` in the menu. Inside of the `production` folder, find the `cache` file and open it up in an editor. This JSON file will have a `COHORTS` section that looks like:

```json
"COHORTS": {
  "iatr_debug_slideshow": {
    "cohort": "B"
  }
}
```

You can either change the cohort directly to see the other content or remove the `iatr_debug_slideshow` entry entirely and refresh the page. It's weighted evenly so there is a 50/50 chance of getting A or B.

If you wish to remove the state related to if the slideshow has been completed you can find it inside of `production/projects/<name-of-project-selected/state.json`. There is a `debugSlideshowComplete` flag that can be removed (Note: the app needs to be restarted for this change to take effect).

### How has the user experience changed?

https://user-images.githubusercontent.com/25158820/220191560-523c75f8-4d67-4611-bc3f-12968b7f0603.mov



### PR Tasks


- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
